starboard/android/shared/application_android.cc

```diff
@@ -208,19 +209,23 @@ bool ApplicationAndroid::DestroyWindow(SbWindow window) {
 
 Event* ApplicationAndroid::WaitForSystemEventWithTimeout(SbTime time) {
   // Limit the polling time in case some non-system event is injected.
-  const int kMaxPollingTimeMillisecond = 10;
+  const int kMaxPollingTimeMillisecond = 1000;
 
   // Convert from microseconds to milliseconds, taking the ceiling value.
   // If we take the floor, or round, then we end up busy looping every time
   // the next event time is less than one millisecond.
-  int timeout_millis = (time + kSbTimeMillisecond - 1) / kSbTimeMillisecond;
+  int timeout_millis =
+      (time <
+       std::min(kSbInt64Max - 1000, 1000 * static_cast<int64_t>(INT_MAX - 1)))
```

Comment:
Instead of `kMaxPollingTimeMillisecond` you probable mean `kSbTimeMillisecond`, and I'd say we shouldn't because `kSbTimeMillisecond` is deprecated in trunk/25.lts and this way the new code is the same as trunk/25.lts.